Library/src/Multiplayer/SpaceEntitySystem.cpp

```diff
@@ -419,6 +409,25 @@ void SpaceEntitySystem::DestroyEntity(SpaceEntity* Entity, CallbackHandler Callb
 
 	CSP_DELETE(Keys);
 
+	csp::common::Array<ComponentUpdateInfo> Info;
+
+	RootHierarchyEntities.RemoveItem(Entity);
+
+	// Manually process the parent updates locally
+	// We want this callback to fire before the deletion so clients can react to children first
```

Comment:
I think this comment needs to be updated as now happens after.

Reply:
It still happens before. It's just outside of the callback now